blackjax/adaptation/window_adaptation.py

```diff
@@ -448,4 +380,17 @@ def stan_warmup_schedule(
         schedule += [(0, False)] * (num_steps - 1 - final_buffer_start)
         schedule.append((0, False))
 
-    return schedule
+    schedule = jnp.array(schedule)
+
+    def schedule_fn(step):
+        """Return the current window and whether this step is the last in the window.
+
+        Passed the supposed last step we stay in the first window.
+
+        """
+        try:
+            return schedule[step]
+        except IndexError:
+            return (0, False)
```

Comment:
Yes and with hindsight I think it makes more sense to `scan` over an array that contains schedule information since the number of steps is known in advance.

Reply:
Yeah I think that should be faster actually.